### DISCUSSION TITLE
Does `app.exit()` not trigger `RunEvent::Exit` event?

### DISCUSSION THREAD

**Discussion by minosss** [tauri-apps/tauri#4662] [Q&A] [answered]
Mac 11.6
tauri 1.0.2

event
```rs
app = tauri::Builder::default()
...
// custom menu item (tray)
SystemTrayEvent::MenuItemClick { id, .. } => match id.as_str() {
  "quit" => {
    app.exit(0)
  },
  _ => {}
}
...
app.run(|handler, event| match event {
  RunEvent::Exit => {
    println!("exit");
  }
})
```

trigger exit event when I quit in the dock bar. but I call `app.exit(0)` in the tray menu, no printing. Is this right?

**Comment by FabianLars** [comment, marked as answer]
> Does app.exit() not trigger RunEvent::Exit event?

Nope. app.exit() calls https://doc.rust-lang.org/std/process/fn.exit.html (after some tauri app specific cleanups), so the EventLoop doesn't explicitly shut down (in a clean way?)
